### PR TITLE
TW-708/preloader+collections

### PR DESCRIPTION
**Pull request by Boltman92**

### CODE REVIEW

**Review by herkoss — APPROVED** [review]

File: src/screens/collection/collection.tsx [hunk 1]
```diff
@@ -77,34 +77,40 @@ export const Collection = () => {
...
-          }
-        }}
-        onEndReachedThreshold={0.5}
-        ListEmptyComponent={emptyComponent}
-        ListFooterComponent={footerComponent}
-        ListHeaderComponent={<View style={styles.emptyBlock} />}
-        windowSize={3}
-        initialNumToRender={3}
-      />
+    <ScrollView style={styles.root} contentContainerStyle={styles.scrollViewContainer}>
```
Comment (lendihop) [inline src/screens/collection/collection.tsx:84]: is this ScrollView necessary? You have a FlatList inside

Comment (Boltman92) [reply src/screens/collection/collection.tsx:84]: yes, bc we need vertical scroll for some types of devices
(we have horizontal Flatlist inside)

Comment (lendihop) [reply src/screens/collection/collection.tsx:84]: ok. did you checked how this works on Android? There could be some issues with gestures

Comment (Boltman92) [reply src/screens/collection/collection.tsx:84]: yes, I checked it, and QA saw it, it works correctly now

Comment (lendihop) [reply src/screens/collection/collection.tsx:84]: ok

**Review by lendihop — APPROVED** [review]

File: src/screens/collection/collection.tsx [hunk 2]
```diff
@@ -41,7 +41,15 @@ export const Collection = () => {
     params.galleryId
   );
 
-  const { setInnerScreenIndex } = useInnerScreenProgress(collectibles.length);
+  const PAGINATION_STEP = useMemo(() => (params.type === ObjktTypeEnum.faContract ? 500 : 15), []);
```
Comment (herkoss) [inline src/screens/collection/collection.tsx:44]: Magic numbers. Fix please here and in all places

**Review by lendihop — APPROVED** [review]

File: src/hooks/use-collectibles-by-collection.hook.ts [hunk 3]
```diff
@@ -21,8 +21,8 @@ export const useCollectibleByCollectionInfo = (
     const subscription = fetchCollectiblesByCollection$(contract, selectedPublicKey, type, offset, galleryId)
       .pipe(
         map(result => result),
-        catchError(err => {
-          showErrorToast({ description: err.message });
+        catchError(() => {
+          showErrorToast({ description: 'Sorry, something went wrong..' });
```
Comment (lendihop) [inline src/hooks/use-collectibles-by-collection.hook.ts:25]: extract error message to a const

**Review by lendihop — APPROVED** [review]

**Review by IrynaKhyzhynska — APPROVED** [review]
android done 1.15_770

**Review by vladvn1905 — APPROVED** [review]
ios done build: 1.15_770

**Pull request merged**